vmdb/app/controllers/api_controller/parser.rb

```diff
@@ -107,6 +107,17 @@ def href_collection_id(path)
       subcollection ? [subcollection.to_sym, s_id] : [collection.to_sym, c_id]
     end
 
+    def parse_id(resource, collection)
+      return nil if resource.blank?
+
+      href = resource["href"]
+      if href.present?
+        return href.match(%r{^.*/#{collection}/[0-9]+$}) ? href.split('/').last : nil
+      end
+
+      resource["id"].present? ? resource["id"] : nil
```

Comment:
Can't this just be `resource["id"]` without the present check, or am I missing something?  Maybe a blank id could come in, but this code doesn't even check for bad values like where id is a bunch of letters.